### DISCUSSION TITLE
Questions about Event Node Priority

### DISCUSSION THREAD

**Discussion by GreatWyrm** [Minestom/Minestom#1667] [Help]
I don't see this in the wiki (and it probably should be documented on the Events page on the wiki), so I have a question on how the event priority works.

So, let's say we have 4 event nodes listening on the same event, with priorities -10, 0, 5, and 10.

Which one will run first and last?
Does a negative number mean it has low priority, and it will run first, or last?

The question I'm trying to get at is what's the general mapping between integer number and order in which it runs, and I'd also like [events wiki page](https://wiki.minestom.net/feature/events) to be updated with this information. The page mentions priority as a component of event nodes, but not what it does.

**Comment by FluxCapacitor2** [comment]
Looking at the [EventNodeImpl](https://github.com/Minestom/Minestom/blob/93718276c54ff4d3dfcc7d410b3fa5b0f8e1024f/src/main/java/net/minestom/server/event/EventNodeImpl.java) class, I believe event nodes with a *lower* event priority number run first, and the node with the highest priority value runs last.

I agree with you about the Wiki. It should probably be documented in the [Javadoc](https://javadoc.minestom.net/net/minestom/server/event/EventNode.html#setPriority(int)) as well.
